### ISSUE TITLE
Theme parent CSS not compiled into right folder (404)?

### ISSUE THREAD

**Issue by rdeelstra** [magento/magento2#2582]
I created two themes for example:

> Example/default
> Example/skin1

The Example/skin1 has as a parent the Example/default. Example/default has no parent (it's a copy of Magento/blank). After I compile it the source code searches for CSS in the Example/skin1 e.g.:
<link  rel="stylesheet" type="text/css"  media="all" href="http://magento2.nl/pub/static/frontend/Example/skin1/nl_NL/mage/calendar.css" />

But skin1 does not have that CSS, and (in my imaginary world) it should get that CSS from the parent if it does not exist in the child theme, or am I wrong?
In the Example/default folder it is compiled and accessible:
http://magento2.nl/pub/static/frontend/Example/default/nl_NL/mage/calendar.css

**Comment by davidalger** [comment]
@rdeelstra When you say "it should get that CSS from the parent" are you talking about links to the static files generating incorrectly on the frontend? I'm trying to understand what you're saying the problem is, or if you're asking a question about how theme fallback works.

**Comment by zanilee** [comment]
Hi, @rdeelstra!

If a _child theme_, Example/skin1, doesn’t have certain css, for example calendar.css, it will look for the css in the parent theme. But then it will copy the css to its own folder, so all child theme static files will be placed to http://magento2.nl/pub/static/frontend/Example/skin1/ folder, and the path to calendar.css would be:
http://magento2.nl/pub/static/frontend/Examle/skin1/nl_NL/mage/calendar.css 

And the _parent theme_, Example/default, has its own static files. It's calendar.css is in http://magento2.nl/pub/static/frontend/Example/default/nl_NL/mage/calendar.css

**Comment by rdeelstra** [comment]
Hi :-)

@davidalger Yes, the static URL is not correct or if I understand @zanilee correctly the parent CSS is not copied to the child directory which results to a 404.

**Comment by zanilee** [comment]
Hi, @rdeelstra!

Could you please provide more detailed steps to reproduce this issue?

**Comment by rdeelstra** [comment]
Hi!
I wanted to try it again but now I ran into a different issue (?) when I login:
Class Magento\Catalog\Api\Data\ProductOptionExtensionFactory does not exist
#0 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/Definition/Runtime.php(44): Magento\Framework\Code\Reader\ClassReader->getConstructor('Magento\Sales\M...')
#1 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php(71): Magento\Framework\ObjectManager\Definition\Runtime->getParameters('Magento\Sales\M...')
#2 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/ObjectManager.php(71): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\Sales\M...')
#3 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/AbstractFactory.php(236): Magento\Framework\ObjectManager\ObjectManager->get('Magento\Sales\A...')
#4 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php(53): Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgument(Array, 'Magento\Sales\A...', NULL, 'orderItemReposi...', 'Magento\Sales\M...')
#5 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php(82): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->_resolveArguments('Magento\Sales\M...', Array, Array)
#6 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/ObjectManager.php(71): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\Sales\M...')
#7 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/AbstractFactory.php(273): Magento\Framework\ObjectManager\ObjectManager->get('Magento\Sales\M...')
#8 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/AbstractFactory.php(249): Magento\Framework\ObjectManager\Factory\AbstractFactory->parseArray(Array)
#9 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php(53): Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgument(Array, NULL, Array, 'relationProcess...', 'OrderRelationsC...')
#10 /Applications/XAMPP/xamppfiles/htdocs/magento2/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php(82): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->_resolveArguments('OrderRelationsC...', Array, Array)
...

So I have to find some time to wrestle with that before I can test it again

**Comment by rdeelstra** [comment]
After clearing the var dir the error above was gone.
I also managed to fix my css 'error'. When I add web/css/default.less in the theme I test everything seems to go fine, without it it seems to bring the error up. Maybe some extra validation and/or improved checkups in this could help new theme developers.
Here is my sample theme for if you would like to test it, default works and child not:
http://mediaride.nl/Testenv2.zip

**Comment by zanilee** [comment]
Hi, @rdeelstra!

We can't reproduce it now. Could you please update your Magento2 installation and try themes again?

**Comment by MomotenkoNatalia** [comment]
@rdeelstra I am closing this issue due to no response, please reopen when you have an update

**Comment by jonsuburb** [comment]
I am experiencing the same issue. I created a child theme (created theme.xml and registration.php).

If I set theme to the child in admin, the page tries to load the CSS files with 404 errors. Its expecting the parent's css files to be copied to the child theme folders in pub/static, but it doesn't exist.

Why are the files not being copied? Is it supposed to happen automatically? Its not behaving as @zanilee described.
